numpy/core/shape_base.py

```diff
@@ -275,3 +278,67 @@ def hstack(tup):
         return _nx.concatenate(arrs, 0)
     else:
         return _nx.concatenate(arrs, 1)
+
+def stack(arrays, axis=0):
+    """
+    Join a sequence of arrays along a new axis.
```

Comment:
Still needs a better explanation of where the new axis is located. It might also be better to follow the list insertion protocol.
```
In [7]: a = [1]*4

In [8]: a.insert(4, 2)

In [9]: a
Out[9]: [1, 1, 1, 1, 2]

In [10]: a = [1]*4

In [11]: a.insert(-1, 2)

In [12]: a
Out[12]: [1, 1, 1, 2, 1]
```